src/components/EditActivityForm.js

```diff
@@ -62,23 +63,27 @@ class EditActivityForm extends Component {
   @action.bound handleAddEventClick(event) {
     this.handleChange();
     event.preventDefault();
-    const date = new Date(this.eventTime + 'T01:01:01.00');
-    if(!isValidDate(date)) {
-      this.invalidDateError = new Error(`Invalid date. Format should be: YYYY-MM-DD`);
-      return;
-    }
-    this.events.push({
-      id: `event-${tempIdCounter++}`,
-      eventTime: date
-    });
-    this.eventTime = null;
+    this.eventTimes.forEach(event => {
+      const date = new Date(event);
+      if (!isValidDate(date)) {
+        this.invalidDateError = new Error(`Invalid date. Format should be: YYYY-MM-DD`);
+        return;
+      }
+      if (this.events.every(event => event.eventTime.getTime() !== date.getTime())) {
+        this.events.push({
+          id: `event-${tempIdCounter++}`,
+          eventTime: date
+        });
+      }
+    })
+    this.eventTimes.splice(0, this.eventTimes.length)
```

Comment:
Call me opinionated, but `this.eventTimes = [];` looks cleaner. This also probably breaks some React state rule (I'm no expert though).

Reply:
Definitely does! Not sure why I cleared the array with `splice()` before.